### PR TITLE
Add dtype and columns type check for shuffle

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

File: dask_expr/tests/test_distributed.py [hunk 1]
```diff
@@ -356,6 +357,19 @@ def test_func(n):
     pd.testing.assert_frame_equal(result, expected)
 
 
+@gen_cluster(client=True)
+async def test_future_in_map_partitions(c, s, a, b):
```
Comment (hendrikmakait) [inline dask_expr/tests/test_distributed.py:361]: The test name seems wrong.

Comment (phofl) [reply dask_expr/tests/test_distributed.py]: very wrong, thx

**Comment by phofl** [comment]
dask/dask ci is expected

**Review by hendrikmakait — APPROVED** [review]
Thanks, @phofl!

**Pull request merged**